### ISSUE TITLE
Multiline PHP in HTML string

### ISSUE THREAD

**Issue by roblourens** [atom/language-php#259]
Kind of related to https://github.com/atom/language-php/issues/246, but from a different angle. If PHP is injected in the middle of an HTML string, the coloring is OK, unless there's a newline in it.

Is this a bug or does the newline change the meaning of the PHP in this case?

The first php line is ok except for inheriting some green from the string, which I think is ok. 

```php
<ol class="
<?php if ($this->getCollapsed() && !$this->getHasSelection()): ?>no-display-current <?php endif ?>
<?php 
if($this->getSingleChoice()) {echo " single-choice ";}?>
<?php if ($this->getSortFeaturedFirst() && 
    in_array($doesnt, $matter)) {} ?>
">
```

![image](https://user-images.githubusercontent.com/323878/29418005-f8b6acf4-831f-11e7-9b53-6f76785330b9.png)

https://github.com/Microsoft/vscode/issues/32700

**Comment by 50Wliu** [comment]
This might be language-html getting very, very confused by the PHP.

**Comment by Ingramz** [comment]
I figured out what causes this.

https://github.com/atom/language-php/blob/efab9c83b91c02f04b8cfd0236e0239e14c4af52/grammars/php.cson#L43

When starting from a new line without leading whitespace, this is a zero width match. Injection selector `L:text.html.php meta.tag` will keep injecting this rule to the left of the list. Because it is being applied recursively without any new characters being matched, it simply bails out.

Easiest way to fix this is simply to 🔥 the rule. However then the question would be whether that rule would be useful if it were fixed.

As far as I know there is no elegant fix for this that would maintain both the intended behavior and allow using injections with `L:` prefix. The best I could do was to borrow `<` character from inside to make it capture at least one character, so it wouldn't get stuck, but then the tag gets chopped like this:
![image](https://user-images.githubusercontent.com/1017132/30069380-4714a54c-9269-11e7-8905-87bc7064f2e5.png)

It might be possible that this can be fixed by telling tokenizer to merge these tokens together (provided there are no ambiguities). Also that `<span class="null">` looks like another bug.
